client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ClusterClientDocumentationIT.java

```diff
@@ -0,0 +1,190 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.documentation;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.LatchedActionListener;
+import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsRequest;
+import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsResponse;
+import org.elasticsearch.client.ESRestHighLevelClientTestCase;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.cluster.routing.allocation.decider.EnableAllocationDecider;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.indices.recovery.RecoverySettings;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * This class is used to generate the Java Cluster API documentation.
+ * You need to wrap your code between two tags like:
+ * // tag::example[]
+ * // end::example[]
+ *
+ * Where example is your tag name.
+ *
+ * Then in the documentation, you can extract what is between tag and end tags with
+ * ["source","java",subs="attributes,callouts,macros"]
+ * --------------------------------------------------
+ * include-tagged::{doc-tests}/ClusterClientDocumentationIT.java[example]
+ * --------------------------------------------------
+ */
+public class ClusterClientDocumentationIT extends ESRestHighLevelClientTestCase {
+
+    public void testClusterPutSettings() throws IOException {
+        RestHighLevelClient client = highLevelClient();
+
+        // tag::put-settings-request
+        ClusterUpdateSettingsRequest request = new ClusterUpdateSettingsRequest();
+        // end::put-settings-request
+
+        // @formatter:off
+        // tag::put-settings-create-settings
+        String transientSettingKey = 
+                RecoverySettings.INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING.getKey();
+        int transientSettingValue = 10;
+        Settings transientSettings = 
+                Settings.builder()
+                .put(transientSettingKey, transientSettingValue, ByteSizeUnit.BYTES)
+                .build(); // <1>
+
+        String persistentSettingKey = 
+                EnableAllocationDecider.CLUSTER_ROUTING_ALLOCATION_ENABLE_SETTING.getKey();
+        String persistentSettingValue = 
+                EnableAllocationDecider.Allocation.NONE.name();
+        Settings persistentSettings = 
+                Settings.builder()
+                .put(persistentSettingKey, persistentSettingValue)
+                .build(); // <2>
+        // end::put-settings-create-settings
+        // @formatter:on
```

Comment:
these are eclipse specific tags right? I don't think we'd want them in our code.

Reply:
According to the docs IntelliJ also supports them.
By default these tags are ignored ( at least for eclipse formatters ). I added them to make clear that these sections intentionally are not following the ES formatting rules.

Should I remove them?

Reply:
yes please I haven't seen them used before in our codebase. At some point we will automate formatting and these classes will have to somehow be ignored I think.